### DISCUSSION TITLE
可以单独安装使用GIE图查询部分么

### DISCUSSION THREAD

**Discussion by bookug** [alibaba/GraphScope#890] [Q&A]
图查询和GAIA的关系是什么？research目录里的gaia可以单独编译使用么？gaia和gaia-x的区别又是啥

**Comment by bookug** [comment]
此外，图查询部分，底层是采用rocksdb存储吧，但是所有图数据又都在内存有一份？所以其实是全内存数据库？rocksdb只是为了批量持久化，而不是像真正的数据库系统那样实时更新到磁盘？

**Comment by longbinlai** [comment]
关于第一个问题，GAIA是图查询的research版本，里面还有一些复杂的问题在解决，所以暂时还没有合并到GraphScope中。Research里面的gaia是可以单独编译的，里面的readme有描述编译和部署流程。Gaia-x是Gaia一次比较大的更新版本，未来会取代gaia，然后进一步合并到整个GraphScope。

**Comment by longbinlai** [comment]
图查询部分，目前GIE里的图数据的确是存储在rocksdb，内存里只是存储了和最近查询相关的cache，而不会保存所有数据；数据是实时更新到磁盘的。
